src/StreamExtended/Network/CustomBufferedStream.cs

```diff
@@ -240,6 +240,30 @@ public async Task<int> PeekByteAsync(int index, CancellationToken cancellationTo
             return streamBuffer[bufferPos + index];
         }
 
+        /// <summary>
+        /// Peeks bytes asynchronous.
+        /// </summary>
+        /// <param name="index">The index.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns></returns>
+        public async Task<byte[]> PeekBytesAsync(int index, int size, CancellationToken cancellationToken = default(CancellationToken))
+        {
+
+            if (Available <= index)
+            {
+                await FillBufferAsync(cancellationToken);
+            }
+
+            if (Available <= (index + size))
```

Comment:
Looking at this again, I believe this code doesn't look right. If index+size is greater than the buffer size (streamBuffer.Length) then you may need to call ResizeBuffer function to increase the buffer size and then call FillBufferAsync. In PeekByte call to this looks like an issue if the buffer is already full.

Reply:
If the assumption is that we should return null if the peek request exceeds the buffer size, then this would be okay. The same is true for PeekByteAsync call (which would return -1 in that case)